### DISCUSSION TITLE
Epp update

### DISCUSSION THREAD

**Discussion by mchamberland** [nrc-cnrc/EGSnrc#1041] [Ideas]
@ftessier I have updated Epp for EGSnrc v2023. I removed the pre-parsing of input files. This means the documentation is now out of date. I don't have the tex files to recreate the user manual, so I wonder if Jonas would still have them?

In any case: what do you think is the best way to share this updated version of Epp? I'm happy to make my repository public. It is by itself, so I think including it as a submodule in EGSnrc might be a good idea?

I'm not 100% convinced I want to be the official maintainer of the code, but also... if Jonas is fine with it, I wouldn't mind too much. I'd need to spend time reading the code to make sure I understand what it's doing!

**Comment by ftessier** [comment]
This is great @mchamberland!

@jlippuner do you by any chance have a copy of the original `tex` files for the documentation? Also, can you confirm that you are ok to release Epp as an EGSnrc submodule or as part of EGSnrc. In the latter case, you retain copyright, but licence becomes AGPL v3. Do you want to release under your username, or are you ok if @mchamberland or someone else maintains it.

@mchamberland, in the affirmative, please give me access to your repo (or make it public!), and let's see. I prefer a submodule with clear ownership or maintainership, but we could also integrate it directly in EGSnrc, especially if it is mostly standalone.

**Comment by mchamberland** [comment]
@ftessier I gave you access to the repo. If we get the OK from @jlippuner, I'll make it public.

**Comment by mchamberland** [comment]
Also, I think the code needs a little further testing, to make sure I didn't break anything. In particular, I need to double-check the dose calculation against egs_dose_scoring, since I had to replace the getVolume() call by getMass() and getting the density.

**Comment by ftessier** [comment]
Sounds good. I am open to integrate it into EGSnrc, since it is "just" a standalone application, if I understand correctly. I agree with testing, but I would release it on `develop` as soon as possible, so others can test it as well!

@jlippuner, I see that the licence is `GPL-2.0-or-later` (SPDX string), which is compatible with EGSnrc's `AGPL-3.0-or-later`. It might be curteous to let the copyright holder **CancerCare Manitoba** know about the release, but strictly speaking it is not necessary.

**Comment by jlippuner** [comment]
This is great news! Thanks for your work @mchamberland! I haven't done anything with Epp or EGSnrc for 10 years, so I'm very happy for someone else to maintain it and for it to be an EGSnrc submodule.

[Epp_manual.zip](https://github.com/nrc-cnrc/EGSnrc/files/12130837/Epp_manual.zip)

Attached are the TEX and word documents I found with the user manual. I *think* the tex file is the latest version. Attached is also some supplemental information that I found that was written by someone else (name of the author is in the document).

Hope this helps!

**Reply by mchamberland** [reply]
@jlippuner thank you very much!
